qiskit/pulse/instructions/call.py

```diff
@@ -38,35 +38,87 @@ def __init__(self, subroutine,
         .. note:: Inline subroutine is mutable. This requires special care for modification.
 
         Args:
-            subroutine (Schedule): A program subroutine to be referred to.
+            subroutine (Union[Schedule, ScheduleBlock]): A program subroutine to be referred to.
             value_dict: Mapping of parameter object to assigned value.
             name: Unique ID of this subroutine. If not provided, this is generated based on
                 the subroutine name.
-        """
-        if name is None:
-            name = f"{self.prefix}_{subroutine.name}"
 
-        super().__init__((subroutine,), None,
-                         channels=tuple(subroutine.channels),
-                         name=name)
-        if value_dict:
-            self.assign_parameters(value_dict)
+        Raises:
+            PulseError: If subroutine is not valid data format.
+        """
+        from qiskit.pulse.schedule import ScheduleBlock, Schedule
+
+        if not isinstance(subroutine, (ScheduleBlock, Schedule)):
+            raise PulseError(f'Subroutine {subroutine} is not acceptable data format.')
+
+        # initialize parameter template
+        # TODO remove self._parameter_table
+        self._arguments = dict()
+
+        if subroutine.is_parameterized():
```

Comment:
So now I'm more confused, because this method is available at the higher level? Maybe because I haven't gotten around to `Schedule` again yet

Reply:
This is because schedule and block still provide user-facing functionalities to deal with parameters. Thus `.parameters`, `.is_parametrized`, `.assign_parameters` are not deprecated in these classes. Actual functionalities are provided by parameter manager attached to these programs.